### DISCUSSION TITLE
+/- lat/lon bug?

### DISCUSSION THREAD

**Discussion by andrew-reiter** [semuconsulting/pynmeagps#54] [General]
New user here; am I missing something??

The sample code and description in the README seem to show automatic North / South, East / West fields based on the sign of the latitude and longitude, but it doesn't seem to be working.

The following script is mostly copy / pasted from the README:
``` python
from pynmeagps import NMEAMessage, GET
msgsp = NMEAMessage('GN', 'GLL', GET, lat=43.123456789, lon=-2.987654321, 
                    status='A', posMode='A', hpnmeamode=0) # standard precision
print(repr(msgsp))
msghp = NMEAMessage('GN', 'GLL', GET, lat=-43.123456789, lon=2.987654321, 
                    status='A', posMode='A', hpnmeamode=1) # high precision
print(repr(msghp))
```
Produces
```
NMEAMessage('GN','GLL', 0, payload=['4307.40741', 'N', '00259.25926', 'E', '202808.32', 'A', 'A'])
NMEAMessage('GN','GLL', 0, payload=['4307.4074073', 'N', '00259.2592593', 'E', '202808.32', 'A', 'A'])
```

.... rather than the `S` and `W` present in the output in the README.

WSL2 / Ubuntu 22.04.3, Python 3.10.12. Library v1.0.34

Happy to convert this to a bug report if need be, but I wanted a sanity check first?

**Comment by semuadmin** [comment]
Hi @andrew-reiter 

I've converted this to an issue #55 as there are actually a couple of problems here.

The documentation needs updating to indicate that the values of `NS` or `EW` are _always_ derived from the sign of `lat` and `lon` when specifying the payload as individual keyword arguments (_so there is never any need to provide `NS` or `EW` values explicitly_). But I've just noticed there is also an issue with message creation with certain combinations of 'NS' and 'EW' values - it's an easy fix which will go into RC 1.0.35.
